src/crewai/crew.py

```diff
@@ -685,6 +703,72 @@ def _run_hierarchical_process(self) -> CrewOutput:
             token_usage=token_usage,
         )
 
+    def _process_async_tasks(
+        self,
+        futures: List[Tuple[Task, Future[TaskOutput], int]],
+        was_replayed: bool = False,
+    ) -> List[TaskOutput]:
+        task_outputs = []
+        for future_task, future, task_index in futures:
+            task_output = future.result()
+            task_outputs.append(task_output)
+            self._process_task_result(future_task, task_output)
+            self._store_execution_log(
+                future_task, task_output, task_index, was_replayed
+            )
+        return task_outputs
+
+    def _find_task_index(
+        self, task_id: str, stored_outputs: List[Dict[str, Any]]
+    ) -> Optional[int]:
+        return next(
+            (
+                index
+                for (index, d) in enumerate(stored_outputs)
+                if d["task_id"] == str(task_id)
+            ),
+            None,
+        )
+
+    def replay_from_task(
```

Comment:
https://www.loom.com/share/b0f489adcaba4ca68c44177bc05a0d46?sid=56d1a54c-1e33-4407-86f0-070984e7d112

I had some questions comparing `replay` to `kickoff`

Also, does replay work with `kickoff_for_each`, `kickoff_async` and `kickoff_for_each_async`?